File: bkmonitor/webpack/src/monitor-api/modules/shield.js
import { request } from '../base';

export const shieldList = request('POST', 'rest/v2/shield/shield_list/');
export const frontendShieldList = request('POST', 'rest/v2/shield/frontend_shield_list/');
export const shieldDetail = request('GET', 'rest/v2/shield/shield_detail/');
export const frontendShieldDetail = request('GET', 'rest/v2/shield/frontend_shield_detail/');
export const frontendCloneInfo = request('GET', 'rest/v2/shield/frontend_clone_info/');
export const shieldSnapshot = request('POST', 'rest/v2/shield/shield_snapshot/');
export const addShield = request('POST', 'rest/v2/shield/add_shield/');
export const bulkAddAlertShield = request('POST', 'rest/v2/shield/bulk_add_alert_shield/');
export const editShield = request('POST', 'rest/v2/shield/edit_shield/');
export const disableShield = request('POST', 'rest/v2/shield/disable_shield/');
export const updateFailureShieldContent = request('GET', 'rest/v2/shield/update_failure_shield_content/');

export default {
  shieldList,
  frontendShieldList,
  shieldDetail,
  frontendShieldDetail,
  frontendCloneInfo,
  shieldSnapshot,
  addShield,
  bulkAddAlertShield,
  editShield,
  disableShield,
  updateFailureShieldContent,
};

File: bkmonitor/kernel_api/views/v4/shield.py
# -*- coding: utf-8 -*-
"""
Tencent is pleased to support the open source community by making 蓝鲸智云 - 监控平台 (BlueKing - Monitor) available.
Copyright (C) 2017-2025 Tencent. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at http://opensource.org/licenses/MIT
Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
"""
from core.drf_resource.viewsets import ResourceRoute, ResourceViewSet
from core.drf_resource import resource


class ShieldViewSet(ResourceViewSet):
    resource_routes = [
        ResourceRoute("POST", resource.shield.add_shield, endpoint="add"),
        ResourceRoute("POST", resource.shield.disable_shield, endpoint="disable"),
        ResourceRoute("POST", resource.shield.edit_shield, endpoint="edit"),
        ResourceRoute("GET", resource.shield.shield_detail, endpoint="detail"),
        ResourceRoute("POST", resource.shield.shield_list, endpoint="search"),
    ]
